### PR TITLE
My Site Dashboard: request cards

### PR DESCRIPTION
**Pull request by leandroalonso**
Part of #17873

This PR hooks up the request to the cards API.

**Important**: the endpoint hasn't been deployed yet, so you'll need to run the diff in your sandbox.

https://user-images.githubusercontent.com/7040243/152412815-ccc54b47-5159-4d3a-b6d6-8adadba6752f.mp4

### Out of scope

Parsing the response or displaying the final loading cells is not part of this task. This is mainly to ensure that the request is working as intended. Those will be tackled in subsequent tasks.

### To test

Make sure the `My Site Dashboard` feature flag is **enabled**.

1. Open the app
2. Tap Dashboard
3. Make sure Quick Links and Posts appear 

## Regression Notes
1. Potential unintended areas of impact
n/a

2. What I did to test those areas of impact (or what existing automated tests I relied on)
n/a

3. What automated tests I added (or what prevented me from doing so)
Added as part of the WPKit PR: https://github.com/wordpress-mobile/WordPressKit-iOS/pull/485

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by leandroalonso** [comment]
@pachlava this task is mainly technical and the UI changes are temporary. I assigned you anyway for visibility, thanks!

File: WordPress/Classes/ViewRelated/Blog/Blog Dashboard/BlogDashboardViewController.swift
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/Blog/Blog Dashboard/BlogDashboardViewController.swift]: Something we may want to address in a future PR:

Could we make this a non-optional `BlogDashboardViewModel`? And maybe we can lazy init the viewModel?

I think we may want to inject a non-optional `Blog` in the init for this VC.

File: WordPress/Classes/ViewRelated/Blog/Blog Dashboard/BlogDashboardViewModel.swift
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/Blog/Blog Dashboard/BlogDashboardViewModel.swift]: What do you think of using guard let to cast `blog.dotComID` as an Int?

**Review by momo-ozawa — COMMENTED** [review]
Works as described, confirmed the new endpoint gets called.

File: WordPress/Classes/ViewRelated/Blog/Blog Dashboard/BlogDashboardViewModel.swift
Comment (leandroalonso) [reply WordPress/Classes/ViewRelated/Blog/Blog Dashboard/BlogDashboardViewModel.swift]: Fixed in 2898f813658cac9559bcd35558e16b7ae41b744e

File: WordPress/Classes/ViewRelated/Blog/Blog Dashboard/BlogDashboardViewController.swift
Comment (leandroalonso) [reply WordPress/Classes/ViewRelated/Blog/Blog Dashboard/BlogDashboardViewController.swift]: Fixed in 2898f813658cac9559bcd35558e16b7ae41b744e

**Comment by jkmassel** [comment]
You can test the <strong>Jetpack</strong> changes on this Pull Request by downloading it from AppCenter <a href='https://install.appcenter.ms/orgs/automattic/apps/jetpack-installable-builds/'>here</a> with build number: <code>pr17889-c45e541</code>. IPA is available <a href='https://appcenter-filemanagement-distrib2ede6f06e.azureedge.net/e3c93efb-1d93-4610-830f-c089e81933d6/Jetpack%20Alpha.ipa?sv=2019-02-02&sr=c&sig=D3JYLRbOJJflXmk8d%2FGhDTYSat1ihtkg6p4mNWXmQgQ%3D&se=2022-02-05T16%3A33%3A44Z&sp=r'>here</a>. If you need access to this, you can ask a maintainer to add you.

**Comment by jkmassel** [comment]
You can test the <strong>WordPress</strong> changes on this Pull Request by downloading it from AppCenter <a href='https://install.appcenter.ms/orgs/automattic/apps/WPiOS-One-Offs/'>here</a> with build number: <code>pr17889-c45e541</code>. IPA is available <a href='https://appcenter-filemanagement-distrib4ede6f06e.azureedge.net/f8fa295e-5212-4cf2-b900-fc617e1c5d37/WordPress%20Alpha.ipa?sv=2019-02-02&sr=c&sig=l9yDNM2DLZwjWrVeum%2BC6LH8c8PsyGH1%2BiHaCDCoMxQ%3D&se=2022-02-05T16%3A34%3A35Z&sp=r'>here</a>. If you need access to this, you can ask a maintainer to add you.

**Review by momo-ozawa — APPROVED** [review]

**Comment by mokagio** [comment]
@momo-ozawa @leandroalonso The `release/19.2` had already been cut from `trunk` before this was merged. If you need these changes in 19.2, please open a cherry-pick PR against `release/19.2`.

Between the [Pending](https://github.com/wordpress-mobile/WordPress-iOS/milestone/41) milestone and the description says "the endpoint hasn't been deployed yet," I guess you don't need this PR in 19.2, but I thought I'd leave a note anyway given the timing of the merge 😄

**Pull request merged**
